src/asciidoc/reference/packages/withkojiDispatch.adoc

```diff
@@ -51,19 +48,177 @@ const dispatch = new Dispatch({
+
+_Object_, contains IDs of currently connected clients as keys and an object of their user data.
+
+==== Example
+[source,json]
+----
+{
+	"0dc15e36-81b4-442b-be4d-bf590ee73c2a": {
+		username: "test1",
+		lastPing: 1600218622595
+	},
+	"1fe2eb80-57e3-435f-9314-faa74853c34a": {
+		username: "test2",
+		lastPing: 1600218623097
+	}
+}
+----
+
 [.hcode, id=".latency", reftext="latency"]
-=== .latency()
+=== .latency
+
+_Number_, the latency of the last message in milliseconds.
+
+[.hcode, id=".shardName", reftext="shardName"]
+=== .shardName
+
+_String_, name of the shard the client is currently connected to.
+
+[.hcode, id=".userInfo", reftext="userInfo"]
+=== .userInfo
+
+_Object_, the user info of the currently connected client.
+
+== Dispatch methods
+
+[.hcode, id=".connect", reftext="connect"]
+=== .connect()
+
+(Async) Connects the dispatch client to an open shard.
+
+==== Example
+
+[source,javascript]
+dispatch.connect();
 
-Checks the latency of dispatch messages.
+[.hcode, id=".disconnect", reftext="disconnect"]
+=== .disconnect()
+
+(Async) Disconnects the dispatch client from the active connection. 
+
+==== Example
+
+[source,javascript]
+dispatch.disconnect();
+
+[.hcode, id=".emitEvent", reftext="emitEvent"]
+=== .emitEvent(eventName, payload)
+
+Emits the named event to all clients on the current shard.
+
+==== Parameters
+
+* `eventName` – _String_, name of the event.
+* `payload` – _Any_, data to send with event.
+
+==== Example
+
+[source,javascript]
+dispatch.emitEvent('myEvent', myDataPayload);
+
+[.hcode, id=".info", reftext="info"]
+=== .info()
+
+Gets info on the shard the client is currently connected to.
 
 ==== Returns
 
-Number, the latency of the last message in milliseconds.
+(Async) _ShardInfo_, Object describing the shard the client is currently connected to with shard name and number of connected clients.
+
+==== Example
+
+[source,javascript]
+----
+dispatch.info.then((info) => {
+    currentInfo = info;
+});
+----
+
+[.hcode, id=".on", reftext="on"]
+=== .on(eventName, handler)
+
+Sets a listener to a specific event and runs the handler when the event is dispatched over the shard.
+
+TIP: For Koji custom events see <<#_dispatch_event, DISPATCH_EVENT>>.
+
+==== Parameters
+
+* `eventName` – _String_, name of the event to subscribe to.
+* `handler` – _Function_, the handler to run when the event is fired.
+
+==== Example
+
+[source,javascript]
+dispatch.on('myEvent', myHandlerFunction);
+
+[.hcode, id=".removeEventListener", reftext="removeEventListener"]
+=== .removeEventListener(eventName)
+
+Removes listeners from the specified event.
+
+==== Parameters
+
+* `eventName` – _String_, name of the event to remove listeners from.
 
 ==== Example
 
 [source,javascript]
-const myLatency = dispatch.latency();
+dispatch.remveEventListener('myEvent');
+
+[.hcode, id=".setUserInfo", reftext="setUserInfo"]
+=== .setUserInfo(userInfo)
+
+Sets the current client's user info for the currently connected shard.
```

Comment:
A bit confusing. Can you clarify? Also, should this cross-reference with the `userInfo` property?
same thing here with spelling out "information" (and any other places I missed in comments :)

Reply:
This sets the user info value and broadcasts to the shard. Perhaps I should remove "client" and just say "Sets the current users's info" to remove confusion between the differences of client and user.